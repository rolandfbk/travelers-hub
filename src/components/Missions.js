import React, { useEffect } from 'react';
import { useSelector, useDispatch, shallowEqual } from 'react-redux';
import { getMissionFromApi, memberLeave, memberJoin } from '../Redux/mission/mission';
import MissionList from './MissionList';

const Missions = () => {
  const listing = useSelector((state) => state.missionReducer, shallowEqual);

  const dispatch = useDispatch();

  const joinMission = (id, join) => {
    if (join) {
      dispatch(memberLeave(id));
    } else {
      dispatch(memberJoin(id));
    }
  };

  useEffect(() => {
    if (listing.length === 0) {
      getMissionFromApi(dispatch);
    }
  }, [dispatch]);

  return (
    <MissionList listing={listing} joinMission={joinMission} />
  );
};

export default Missions;
