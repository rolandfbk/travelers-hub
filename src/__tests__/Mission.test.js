import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../components/Missions';
import store from '../Redux/Store/ConfigureStore';

const MissionProvider = () => (
  <Provider store={store}>
    <Missions />
  </Provider>
);

it('renders correctly', () => {
  const tree = renderer.create(<MissionProvider />).toJSON();
  expect(tree).toMatchSnapshot();
});
