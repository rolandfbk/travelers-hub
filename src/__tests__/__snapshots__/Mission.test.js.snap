// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="mission-section"
>
  <table
    className="mission-table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
          Â 
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</section>
`;
